### ISSUE TITLE
Hanging at 50% when searching for QR Code

### ISSUE THREAD

**Issue by sethpjohnson** [rsbauer/gps2photos#1]
Not sure what I can do to help. Seems to stick there and does nothing.

**Comment by rsbauer** [comment]
Sorry to hear it's not working.  When you run it, does "gps4camera QR code image" show a green checkmark?  Or is it showing "?".  

What format is the images?  If raw, from which camera brand?

**Comment by rsbauer** [comment]
Also, I'd highly recommend making copied of the images and putting them somewhere safe.  Using a copy, I'd give gps2photos 3 files:

1. The GPX file
2. The image file with the QR code
3. 1 image file you wish to tag

Then run gps2photos and see if it still hangs.

**Comment by sethpjohnson** [comment]
It still shows the `?` while running. The file format is AWR (Raw) from Sony a6400.

I'll do this test when I get home :)

**Comment by rsbauer** [comment]
I only have a Nikon and I'm relying on the Mac OS to provide image reading support.  I'm wondering if by default, the Mac isn't able to process the AWR format out of the box.  You can test this by:

1. Convert the AWR image with the QR code to something like a JPG or PNG
2. Try rescanning with the GPX file and 1 image
3. See if "gps4camera QR code image" changes from ? to a green check mark

If you get a green check mark, then it's likely the AWR format is a bit problematic for the Mac and the app.  Although, I do suspect if exiftool is able to read the AWR format, then processing should work and you should find the 1 image updated with the correct time and a geo tag.  

If you do not get a green check mark when using a JPG or PNG for the QR code image, then I would suggest retaking an image of the QR code (try to fill the photo frame with your device's screen as much as possible).  

Please let me know which path works out for you.  I'm curious about the AWR format and if that might be causing an issue.

**Comment by sethpjohnson** [comment]
3 file test (image, QR, gpx). 

- Fails with AWR QR image. ❌
- Converted AWR to JPG and it finds the QR code. ✅
  - Correct GPS coordinates to the AWR file. ✅

At least I have a good work around. 😄

**Comment by rsbauer** [comment]
Glad to hear you were able to get it to work!  Bummer the app doesn't support AWR.  Do you mind posting the AWR file with the QR code?  If you can, I'll investigate getting gps2photos to process it.

**Comment by sethpjohnson** [comment]
https://d.pr/f/OqtZF8

There you go!

**Comment by rsbauer** [comment]
Thank you!

I've poked around and I'm struggling to find a way to read the ARW format.  I've tried a couple different methods around NSImage and CGImageSourceCreateWithData (can't even seem to get the metadata from the image with this).  

Short term, the app needs to provide feedback to the user letting them know the app failed to read the image.  Longer term, it would be nice if the app could read ARW files.
